src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala

```diff
@@ -40,6 +40,20 @@ import java.time.{Duration, Period}
 class FeelInterpreter {
 
   def eval(expression: Exp)(implicit context: EvalContext): Val = {
+    if (Thread.interrupted()) {
+      throw new InterruptedException()
+    }
+
+    val result = evalInternal(expression)
+    result match {
+      case ErroredVal(value, _) =>
+        value
+      case _ => result
+    }
```

Comment:
:wrench: Having two different evaluation methods with the same signature makes the code harder to understand and could lead to issues.